src/pipx/commands/inject.py

```diff
@@ -32,7 +32,7 @@ def inject_dep(
 ) -> bool:
     logger.debug("Injecting package %s", package_spec)
 
-    if not venv_dir.exists() or not next(venv_dir.iterdir()):
+    if not venv_dir.exists() or next(venv_dir.iterdir(), None) is None:
```

Comment:
`venv_dir.iterdir()` yields `Path` which does not implement `__bool__` or `__len__` so it will always be true in boolean context. We might want to defend against `StopIteration` exceptions raised by `next()`.